### PR TITLE
Fix lints

### PR DESCRIPTION
**Pull request by Hoverbear**
Fixes some clippy lints.

Affected: 

* #161
* #159 
* #160 
* #162

### CODE REVIEW

File: src/raft_log.rs [hunk 1]
```diff
@@ -651,7 +651,7 @@ mod test {
         let unstable_index = 750u64;
         let last_term = last_index;
         let storage = MemStorage::new();
-        for i in 1..(unstable_index + 1) {
+        for i in 1..=unstable_index + 1 {
```
Comment (overvenus) [inline src/raft_log.rs]: Remove `+1`?

Comment (Hoverbear) [reply src/raft_log.rs]: Oh no, thanks for noticing.

File: src/raft_log.rs [hunk 2]
```diff
@@ -1312,7 +1312,7 @@ mod test {
 
         for (i, &(last_index, ref compact, ref wleft, wallow)) in tests.iter().enumerate() {
             let store = MemStorage::new();
-            for i in 1u64..(last_index + 1) {
+            for i in 1u64..=(last_index) {
```
Comment (BusyJay) [inline src/raft_log.rs]: `()` is redundant.

**Comment by Hoverbear** [comment]
The `cargo fmt` version changed due to staleness so I had to make a new commit. :( PTAL again.

**Comment by Hoverbear** [comment]
Ping

**Pull request merged**